### PR TITLE
Enable load from FSI and Categories support

### PR DESCRIPTION
**Pull request by vnlitvinov**

### CODE REVIEW

File: ibis/omniscidb/client.py [hunk 1]
```diff
@@ -182,16 +165,22 @@ def from_ibis(cls, dtype, nullable=None):
             if the given data type was not implemented.
         """
         dtype_ = type(dtype)
-        if dtype_ in cls.ibis_dtypes:
-            typename = cls.ibis_dtypes[dtype_]
-        elif dtype in cls.ibis_dtypes:
-            typename = cls.ibis_dtypes[dtype]
+        for t in (type(dtype), dtype):
```
Comment (anmyachev) [inline ibis/omniscidb/client.py]: maybe `type` instead of `t` ?

File: ibis/omniscidb/operations.py [hunk 2]
```diff
@@ -57,6 +58,12 @@ def _cast(translator, expr):
     arg, target = op.args
     arg_ = translator.translate(arg)
 
+    if isinstance(arg.type(), dt.Category) and isinstance(target, dt.Integer):
+        arg_ = 'key_for_string({0!s})'.format(arg_)
```
Comment (anmyachev) [inline ibis/omniscidb/operations.py]: What means `key_for_string` ?
Shouldn't we return something like `CAST (... AS ...)`?

File: ibis/omniscidb/client.py [hunk 3]
```diff
@@ -7,7 +7,8 @@
 import regex as re
 from pymapd._parsers import _extract_column_details
 from pymapd.cursor import Cursor
-from pymapd.dtypes import TDatumType as pymapd_dtype
+from pymapd.dtypes import TDatumType as pymapd_datumtype
```
[the remark this answers is not in the document]
Comment (vnlitvinov) [reply ibis/omniscidb/client.py]: To not confuse with `pymapd.dtypes` name. Also to not confuse with next import :)

File: ibis/omniscidb/client.py [hunk 4]
```diff
@@ -1062,7 +1069,7 @@ def database(self, name=None):
                 protocol=self.protocol,
                 session_id=self.session_id,
                 ipc=self.ipc,
-                device=self.device,
+                gpu_device=self.gpu_device,
```
[the remark this answers is not in the document]
Comment (vnlitvinov) [reply ibis/omniscidb/client.py]: I just did a PR from my "feature branch", I think github should be smart enough about figuring it out.

File: ibis/omniscidb/client.py (hunk 1 above)
Comment (vnlitvinov) [reply ibis/omniscidb/client.py]: `type` is a builtin Python function; usually it's a bad idea to override builtin functions :)
I can use a `type_`, but I'm not sure if that would really benefit the readability here.

File: ibis/omniscidb/operations.py (hunk 2 above)
Comment (vnlitvinov) [reply ibis/omniscidb/operations.py]: We cannot. This is building the SQL which OmniSci has to understand, and `key_for_string()` is their function which, when called, produces the index number behind the dictionary encoding text in the column.

**Pull request merged**